src/Explorer/Panes/PublishNotebookPane/PublishNotebookPane.tsx

```diff
@@ -74,6 +79,14 @@ export const PublishNotebookPane: FunctionComponent<PublishNotebookPaneAProps> =
     setContent(newContent);
   }, []);
 
+  useEffect(() => {
+    setImageSrc(notebookSnapshot);
+  }, [notebookSnapshot]);
+
+  useEffect(() => {
+    setFormError(notebookSnapshotError);
+  }, [notebookSnapshotError]);
```

Comment:
This feels strange to me to have to set these inside a useEffect. But I am not exactly sure what to do differently. Maybe there is a larger state issue to fix that is out of scope for this PR :/

Reply:
I admit that was the easiest way to transcribe the current behavior ("if one of these values change, update the corresponding state"), but could be overkill.
Conceptually, `imageSrc` and these zustand store state variables are both state variables and the true react way is probably to not update a state variable with another variable, but instead somehow compose them together in an expression at the time of use. Something like:
` const response = await junoClient.publishNotebook( .... , notebookSnapshot ?? imageSrc, ....)`

But yes, I'd prefer getting this in and iterate :).